File: rules/module-naming-consistent/evals/cases/user_repository.ts
import type { Pool } from "pg";
import type { User, Order, Cart, Invoice } from "../model.ts";

export async function getUser(db: Pool, id: string): Promise<User | null> {
  const { rows } = await db.query<User>("select * from users where id = $1", [id]);
  return rows[0] ?? null;
}

export async function fetchOrder(db: Pool, id: string): Promise<Order | null> {
  const { rows } = await db.query<Order>("select * from orders where id = $1", [id]);
  return rows[0] ?? null;
}

export async function loadCart(db: Pool, id: string): Promise<Cart | null> {
  const { rows } = await db.query<Cart>("select * from carts where id = $1", [id]);
  return rows[0] ?? null;
}

export async function retrieveInvoice(db: Pool, id: string): Promise<Invoice | null> {
  const { rows } = await db.query<Invoice>("select * from invoices where id = $1", [id]);
  return rows[0] ?? null;
}

File: rules/python/module-name-describes-contents/fixtures/user_repository.py
import sqlite3
from dataclasses import dataclass


@dataclass
class User:
    id: int
    email: str


@dataclass
class Order:
    id: int
    user_id: int
    total_cents: int


class UserRepository:
    def __init__(self, conn: sqlite3.Connection) -> None:
        self.conn = conn

    def find_user(self, user_id: int) -> User | None:
        row = self.conn.execute("SELECT id, email FROM users WHERE id = ?", (user_id,)).fetchone()
        return User(*row) if row else None

    def find_order(self, order_id: int) -> Order | None:
        row = self.conn.execute("SELECT id, user_id, total_cents FROM orders WHERE id = ?", (order_id,)).fetchone()
        return Order(*row) if row else None

    def orders_for_cart(self, cart_id: int) -> list[Order]:
        rows = self.conn.execute("SELECT id, user_id, total_cents FROM orders WHERE cart_id = ?", (cart_id,)).fetchall()
        return [Order(*r) for r in rows]

    def unpaid_invoices(self) -> list[int]:
        rows = self.conn.execute("SELECT id FROM invoices WHERE paid = 0").fetchall()
        return [r[0] for r in rows]
